daemon/container_operations_windows.go

```diff
@@ -35,6 +43,62 @@ func detachMounted(path string) error {
 	return nil
 }
 
+func (daemon *Daemon) setupSecretDir(c *container.Container) (setupErr error) {
+	if len(c.SecretReferences) == 0 {
+		return nil
+	}
+
+	localMountPath := c.SecretMountPath()
+	logrus.Debugf("secrets: setting up secret dir: %s", localMountPath)
+
+	defer func() {
+		if setupErr != nil {
+			if err := os.RemoveAll(localMountPath); err != nil {
+				log.Errorf("error cleaning up secret mount: %s", err)
+			}
+		}
+	}()
+
+	// create secrets root
+	system.MkdirAll(localMountPath, 0)
+
+	for _, s := range c.SecretReferences {
```

Comment:
I meant.... in similar situations, we've kept the same code common, and factored out the differences. There's a lot of overlap here. I can see maintenance being done on one side and forgotten on the other as it currently stands.